yb-voyager/cmd/import.go

```diff
@@ -399,14 +397,19 @@ func validateFFDBSchemaFlag() {
 }
 
 func validateParallelismFlags() {
-	if tconf.EnableYBAdaptiveParallelism {
-		if tconf.Parallelism > 0 {
-			utils.ErrExit("Error: --parallel-jobs flag cannot be used with --enable-adaptive-parallelism flag")
+	if importerRole == TARGET_DB_IMPORTER_ROLE || importerRole == IMPORT_FILE_ROLE {
+		if tconf.EnableYBAdaptiveParallelism {
+			if tconf.Parallelism > 0 {
+				utils.ErrExit("Error: --parallel-jobs flag cannot be used with --enable-adaptive-parallelism flag")
+			}
 		}
-	}
-	if tconf.MaxParallelism > 0 {
-		if !tconf.EnableYBAdaptiveParallelism {
-			utils.ErrExit("Error: --adaptive-parallelism-max flag can only be used with --enable-adaptive-parallelism true")
+		if tconf.MaxParallelism > 0 {
+			if !tconf.EnableYBAdaptiveParallelism {
+				utils.ErrExit("Error: --adaptive-parallelism-max flag can only be used with --enable-adaptive-parallelism true")
+			}
 		}
+	} else {
+		// not relevant for source-replica and source
+		tconf.EnableYBAdaptiveParallelism = false
 	}
```

Comment:
should we add a console message saying not relevant.

Reply:
Ah, so the thing is that the flag `enable-adaptive-parallelsim` isn't available for import-data-to-source-replica/source. It's just because the default is now `true`, i'm just setting it to false in those scenarios. Since it's the same flag, there isn't a nice way to choose different defaults based on mode.